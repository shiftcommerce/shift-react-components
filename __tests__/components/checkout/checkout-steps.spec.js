// Libraries
import React from 'react'

// Components
import CheckoutSteps from '../../../src/components/checkout/checkout-steps'

test('Renders steps correctly', () => {
  // Act
  const wrapper = shallow(<CheckoutSteps currentStep={1} />)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Shipping Address')
  expect(wrapper).toIncludeText('Shipping Method')
  expect(wrapper).toIncludeText('Payment')
  expect(wrapper).toIncludeText('Review & Submit')
})

test('Renders current step 1 correctly', () => {
  // Act
  const wrapper = shallow(<CheckoutSteps currentStep={1} />)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('.c-step-indicator--active')).toIncludeText('Shipping Address')
})

test('Renders current step 2 correctly', () => {
  // Act
  const wrapper = shallow(<CheckoutSteps currentStep={2} />)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('Link').length).toEqual(1)
  expect(wrapper.find('.c-step-indicator--active')).toIncludeText('Shipping Method')
})

test('Renders current step 3 correctly', () => {
  // Act
  const wrapper = shallow(<CheckoutSteps currentStep={3} />)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('Link').length).toEqual(2)
  expect(wrapper.find('.c-step-indicator--active')).toIncludeText('Payment')
})

test('Renders current step 4 correctly', () => {
  // Act
  const wrapper = shallow(<CheckoutSteps currentStep={4} />)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('Link').length).toEqual(3)
  expect(wrapper.find('.c-step-indicator--active')).toIncludeText('Review & Submit')
})

test('Provides onClick handlers to links', () => {
  // Act
  const wrapper = mount(<CheckoutSteps
    currentStep={4}
    stepActions={{
      3: () => 'onClick test'
    }}
  />)

  // Assert
  expect(wrapper.find('a').at(2).props().onClick()).toEqual('onClick test')
})
