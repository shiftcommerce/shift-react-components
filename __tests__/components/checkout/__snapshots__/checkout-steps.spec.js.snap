// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders current step 1 correctly 1`] = `
<div
  className="o-header--checkout"
>
  <div
    className="c-checkout__steps"
  >
    <div
      className="c-step-indicators"
    >
      <div
        className="c-step-indicators__steps"
      >
        <div
          className="o-header__step"
          key="0"
        >
          <div
            className="c-step-indicator c-step-indicator--active"
          >
            <div
              className="c-step-indicator__position c-step-indicator__position--active"
            >
              1
            </div>
            <div
              className="c-step-indicator__line c-step-indicator__line--active"
            />
            <div
              className="c-step-indicator__subtitle c-step-indicator__subtitle--active"
            >
              Payment Method
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="1"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              2
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Shipping Address
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="2"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              3
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Shipping Method
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="3"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              4
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Payment
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="4"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              5
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Review & Submit
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders current step 2 correctly 1`] = `
<div
  className="o-header--checkout"
>
  <div
    className="c-checkout__steps"
  >
    <div
      className="c-step-indicators"
    >
      <div
        className="c-step-indicators__steps"
      >
        <div
          className="o-header__step"
          key="0"
        >
          <Link
            href="/checkout/payment-method"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Payment Method
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="1"
        >
          <div
            className="c-step-indicator c-step-indicator--active"
          >
            <div
              className="c-step-indicator__position c-step-indicator__position--active"
            >
              2
            </div>
            <div
              className="c-step-indicator__line c-step-indicator__line--active"
            />
            <div
              className="c-step-indicator__subtitle c-step-indicator__subtitle--active"
            >
              Shipping Address
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="2"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              3
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Shipping Method
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="3"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              4
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Payment
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="4"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              5
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Review & Submit
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders current step 3 correctly 1`] = `
<div
  className="o-header--checkout"
>
  <div
    className="c-checkout__steps"
  >
    <div
      className="c-step-indicators"
    >
      <div
        className="c-step-indicators__steps"
      >
        <div
          className="o-header__step"
          key="0"
        >
          <Link
            href="/checkout/payment-method"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Payment Method
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="1"
        >
          <Link
            href="/checkout/shipping-address"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Shipping Address
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="2"
        >
          <div
            className="c-step-indicator c-step-indicator--active"
          >
            <div
              className="c-step-indicator__position c-step-indicator__position--active"
            >
              3
            </div>
            <div
              className="c-step-indicator__line c-step-indicator__line--active"
            />
            <div
              className="c-step-indicator__subtitle c-step-indicator__subtitle--active"
            >
              Shipping Method
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="3"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              4
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Payment
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="4"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              5
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Review & Submit
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders current step 4 correctly 1`] = `
<div
  className="o-header--checkout"
>
  <div
    className="c-checkout__steps"
  >
    <div
      className="c-step-indicators"
    >
      <div
        className="c-step-indicators__steps"
      >
        <div
          className="o-header__step"
          key="0"
        >
          <Link
            href="/checkout/payment-method"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Payment Method
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="1"
        >
          <Link
            href="/checkout/shipping-address"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Shipping Address
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="2"
        >
          <Link
            href="/checkout/shipping-method"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Shipping Method
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="3"
        >
          <div
            className="c-step-indicator c-step-indicator--active"
          >
            <div
              className="c-step-indicator__position c-step-indicator__position--active"
            >
              4
            </div>
            <div
              className="c-step-indicator__line c-step-indicator__line--active"
            />
            <div
              className="c-step-indicator__subtitle c-step-indicator__subtitle--active"
            >
              Payment
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="4"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              5
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Review & Submit
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders current step 5 correctly 1`] = `
<div
  className="o-header--checkout"
>
  <div
    className="c-checkout__steps"
  >
    <div
      className="c-step-indicators"
    >
      <div
        className="c-step-indicators__steps"
      >
        <div
          className="o-header__step"
          key="0"
        >
          <Link
            href="/checkout/payment-method"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Payment Method
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="1"
        >
          <Link
            href="/checkout/shipping-address"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Shipping Address
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="2"
        >
          <Link
            href="/checkout/shipping-method"
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Shipping Method
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="3"
        >
          <Link
            as="/checkout/payment"
            href="/checkout/payment"
            shallow={true}
          >
            <div
              className="c-step-indicator c-step-indicator--completed"
            >
              <div
                className="c-step-indicator__position c-step-indicator__position--completed"
              >
                <Image
                  className="c-step-indicator__icon"
                  src={Object {}}
                />
              </div>
              <div
                className="c-step-indicator__line c-step-indicator__line--completed"
              />
              <div
                className="c-step-indicator__subtitle c-step-indicator__subtitle--completed"
              >
                Payment
              </div>
            </div>
          </Link>
        </div>
        <div
          className="o-header__step"
          key="4"
        >
          <div
            className="c-step-indicator c-step-indicator--active"
          >
            <div
              className="c-step-indicator__position c-step-indicator__position--active"
            >
              5
            </div>
            <div
              className="c-step-indicator__line c-step-indicator__line--active"
            />
            <div
              className="c-step-indicator__subtitle c-step-indicator__subtitle--active"
            >
              Review & Submit
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders steps correctly 1`] = `
<div
  className="o-header--checkout"
>
  <div
    className="c-checkout__steps"
  >
    <div
      className="c-step-indicators"
    >
      <div
        className="c-step-indicators__steps"
      >
        <div
          className="o-header__step"
          key="0"
        >
          <div
            className="c-step-indicator c-step-indicator--active"
          >
            <div
              className="c-step-indicator__position c-step-indicator__position--active"
            >
              1
            </div>
            <div
              className="c-step-indicator__line c-step-indicator__line--active"
            />
            <div
              className="c-step-indicator__subtitle c-step-indicator__subtitle--active"
            >
              Payment Method
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="1"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              2
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Shipping Address
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="2"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              3
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Shipping Method
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="3"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              4
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Payment
            </div>
          </div>
        </div>
        <div
          className="o-header__step"
          key="4"
        >
          <div
            className="c-step-indicator"
          >
            <div
              className="c-step-indicator__position"
            >
              5
            </div>
            <div
              className="c-step-indicator__line"
            />
            <div
              className="c-step-indicator__subtitle"
            >
              Review & Submit
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
